### PR TITLE
Template for before/after build actions

### PR DESCRIPTION
**Pull request by NN---**
#60

### CODE REVIEW

File: src/NuProj.Targets/NuProj.targets [hunk 1]
```diff
@@ -1031,9 +1040,35 @@
...
+    Redefine this target in your project in order to run tasks just after Clean.
+    ============================================================
+    -->
+  <Target Name="AfterClean"/>
+  
+  <PropertyGroup>
+    <BuildDependsOn>
+      BeforeBuild;
+      CreatePackage;
+      AfterBuild
```
Comment (AArnott) [inline src/NuProj.Targets/NuProj.targets]: Why is `AfterBuild` listed here? That would make it run *before* Build.

File: src/NuProj.ProjectSystem.12/Templates/Projects/NuProj/NuProj.nuproj [hunk 2]
```diff
@@ -32,4 +32,10 @@
     <Content Include="Readme.txt" />
   </ItemGroup>
   <Import Project="$(NuProjPath)\NuProj.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+  <Target Name="BeforeBuild">
+  </Target>  
+  <Target Name="AfterBuild">
+  </Target>
+  -->
```
Comment (AArnott) [inline src/NuProj.ProjectSystem.12/Templates/Projects/NuProj/NuProj.nuproj]: Personally I'm not a fan of this extra xml. It trails most projects and most projects don't do anything with them. Can we leave this off the template?

**Comment by AArnott** [comment]
The change mostly looks good, aside from my remarks. 
The `*DependsOn` property is fine, no objections.
But defining empty targets for `BeforeBuild` and `AfterBuild` artificially feels like an obsolete pattern since MSBuild has supported `AfterTargets` and `BeforeTargets` for quite some time and seems to fill this need.

File: src/NuProj.ProjectSystem.12/Templates/Projects/NuProj/NuProj.nuproj (hunk 2 above)
Comment (NN---) [inline src/NuProj.ProjectSystem.12/Templates/Projects/NuProj/NuProj.nuproj]: I guess the default project template has it to make it clear that you can use it.
If you think it is unneeded I can remove the comment.

Comment (AArnott) [inline src/NuProj.ProjectSystem.12/Templates/Projects/NuProj/NuProj.nuproj]: I agree with the purpose. I just feel like it contributes to noise for almost everyone, and the few who want to customize the build *should* know enough about MSBuild to not need the hint. The cost outweighs the benefits by my estimation.

**Comment by NN---** [comment]
Fixed according to the review.

**Comment by NN---** [comment]
The DependsOn pattern is copied from Microsoft.Common.CurrentVersion.targets .

File: src/NuProj.Targets/NuProj.targets [hunk 3]
```diff
@@ -1009,7 +1009,16 @@
 
     =================================================================================================================== -->
 
-  <Target Name="Clean" DependsOnTargets="AssignProjectConfiguration;EstablishNuGetPaths">
+  <PropertyGroup>
+    <CleanDependsOn>
+      BeforeClean;
+      AssignProjectConfiguration;
+      EstablishNuGetPaths
```
Comment (AArnott) [inline src/NuProj.Targets/NuProj.targets]: Missing a semicolon at the end of this line.

Comment (NN---) [inline src/NuProj.Targets/NuProj.targets]: Fixed.
I don't see ending semicolons in any place in this file, and I don't think it is relevant to this PR.

For instance:
```
DependsOnTargets="AssignNuProjPackageDependenciesTargetFramework;GetNuGetPackageDependencies"
```

**Comment by AArnott** [comment]
Thanks for your contribution.

**Comment by NN---** [comment]
Thanks.
Are you going to update vsix in the gallery ?

**Comment by AArnott** [comment]
I can't. I don't have permissions to do so. @terrajobst does it pretty infrequently. I never use the MSI from the gallery anyway except for IDE support. I use the build authoring that comes from the NuProj package on nuget.org. I can't update that either, so I maintain a myget.org feed with the latest package from this project that I need.

**Comment by NN---** [comment]
Maybe you should create a package of yourself which you can update more frequently and call it 'Beta' or smth like that.

**Comment by AArnott** [comment]
That's exactly what I do. 
For instance today I pushed a NuProj package up (version 0.10.18-beta-gcd398090a4)
You can get to it by adding this to your nuget package sources: https://www.myget.org/F/aarnott/api/v3/index.json

**Pull request merged**